src/wallet/mod.rs

```diff
@@ -983,6 +983,59 @@ impl Wallet {
         tracing::trace!("Trusted pending: {}", balance.trusted_pending);
         Ok(balance)
     }
+ 
+    pub async fn list_wallet_transactions(&self) -> Result<Vec<WalletTransaction>> {
+        let wallet = self.bitcoin_wallet.lock();
+        let transactions = wallet.transactions();
+        let mut txs = Vec::new();
+        
+        for tx in transactions {
+            // Calculate total input value
+            let mut input_value = 0;
+            for input in tx.tx_node.tx.input.iter() {
+                if let Ok(prev_output) = self.bitcoin_blockchain.fetch_tx(input.previous_output.txid) {
+                    input_value += prev_output.output[input.previous_output.vout as usize].value.to_sat();
+                }
+            }
+            
+            // Calculate total output value
+            let output_value: u64 = tx.tx_node.tx.output.iter().map(|o| o.value.to_sat()).sum();
+
+            // Fee is difference between inputs and outputs
+            let fee_sats = input_value.saturating_sub(output_value);
+
+            // Calculate received/sent by checking if outputs/inputs belong to wallet
+            let mut received_sats = 0;
+            let mut sent_sats = 0;
+            
+            // Check outputs to wallet
+            for output in tx.tx_node.tx.output.iter() {
+                if wallet.is_mine(output.script_pubkey.clone()) {
+                    received_sats += output.value.to_sat();
+                }
+            }
+            
+            // Check inputs from wallet 
+            for input in tx.tx_node.tx.input.iter() {
+                if let Some(prev_output) = wallet.get_tx(input.previous_output.txid) {
```

Comment:
Same here as above, is this an I/O operation?

Reply:
Yes, this is an i/o operation! I'm currently running it in signet, and the call to listtransaction is as fast as CreateNewAddress

```
Executed in   39.73 millis    fish           external
   usr time   18.41 millis    0.09 millis   18.31 millis
   sys time   10.75 millis    1.08 millis    9.67 millis
```